services/transactionService.ts

```diff
@@ -0,0 +1,507 @@
+import Web3 from 'web3'
+import abiDecoder from 'abi-decoder'
+import { findTokenByNetworkAndSymbol, NETWORK_IDS } from '../utils/tokenUtils';
+import { errorMessages } from '../utils/errorMessages';
+import { NetworkTransactionInfo, TransactionDetailInput } from '../types/TransactionInquiry';
+import axios from 'axios';
+import config from '../config';
+
+const mainnetNodeUrl = config.get('MAINNET_NODE_HTTP_URL') as string
+const mainnetWeb3 = new Web3(mainnetNodeUrl)
+const ropstenNodeUrl = config.get('ROPSTEN_NODE_HTTP_URL') as string
+const ropstenWeb3 = new Web3(ropstenNodeUrl)
+const xdaiWeb3NodeUrl = config.get('XDAI_NODE_HTTP_URL') as string
+const xdaiWeb3 = new Web3(xdaiWeb3NodeUrl)
+
+const getNetworkWeb3 = (networkId: number): Web3 => {
+  switch (networkId) {
+    case  NETWORK_IDS.MAIN_NET:
+      return mainnetWeb3
+
+    case  NETWORK_IDS.ROPSTEN:
+      return ropstenWeb3
+
+    case  NETWORK_IDS.XDAI:
+      return xdaiWeb3
+    default:
+      throw new Error(errorMessages.INVALID_NETWORK_ID)
+  }
+
+}
+
+/**
+ * @see @link{https://gist.github.com/veox/8800debbf56e24718f9f483e1e40c35c, https://stackoverflow.com/a/67617052/4650625}
+ */
+const erc20ABI =
```

Comment:
Please put it somewhere else, a separate file in asset folder or something like taht.